public/addEditMovie.js

```diff
@@ -0,0 +1,121 @@
+import { enableInput, inputEnabled, message, setDiv, token } from "./index.js";
+import { showMovie } from "./movie.js";
+
+let addEditDiv = null;
+let director = null;
+let title = null;
+let genre = null;
+let addingMovie = null;
+
+export const handleAddEditM = () => {
+  addEditDiv = document.getElementById("edit-movie");
+  director = document.getElementById("director");
+  title = document.getElementById("title");
+  genre = document.getElementById("genre");
+  addingMovie = document.getElementById("adding-movie");
+  const editCancel = document.getElementById("edit-cancel");
+
+  addEditDiv.addEventListener("click", async (e) => {
+    if (inputEnabled && e.target.nodeName === "BUTTON") {
+      if (e.target === addingMovie) {
+        enableInput(false);
+
+        let method = "POST";
+        let url = "/api/v1/movie";
+
+        if (addingMovie.textContent === "update") {
+          method = "PATCH";
+          url = `/api/v1/movie/${addEditDiv.dataset.id}`;
+        }
+        console.log("URL de la solicitud", url);
+        try {
+          const response = await fetch(url, {
+            method: method,
+            headers: {
+              "Content-Type": "application/json",
+              Authorization: `Bearer ${token}`,
+            },
+            body: JSON.stringify({
+              director: singer.value,
```

Comment:
this will need to be `director.value` - needs to match variable name defined on line 12 :)